### DISCUSSION TITLE
Framework agnostic bootstrapping of Resilience4j from properties for non-spring Java Applications

### DISCUSSION THREAD

**Discussion by deecodeme** [resilience4j/resilience4j#1962] [Ideas]
Hi,

I am a super fan of resilience4j's lightweight and framework-agnostic implementation. I also like the Resilience4j bootstrapping(Registries initialization) for spring and spring-boot applications from properties.

I wanted to extend the same concept for non-spring java applications and have a library called `resilinence4j-config` which shall bootstrap Resilience4j from Properties. This is to have a framework agnostic Resilience4j bootstrapping from configurations.

Can you please be kind to help me with below two questions?
1. Is there a library already doing what I mentioned as the problem statement and solution above?
2. If it does not exist, can I contribute a module called `resilinence4j-config`?

**Comment by RobWin** [comment]
Hi,

someone tried to create a module based on commons-configuration2. But it wasn't merged.

https://github.com/resilience4j/resilience4j/pull/996

**Comment by deecodeme** [comment]
Thanks @RobWin ,

I will go through the previous attempt and see what needs to be done to make it work.

**Comment by deecodeme** [comment]
Hi @RobWin

I have raised the pull request https://github.com/resilience4j/resilience4j/pull/1991. Can you please review and provide your valuable suggestions and concerns?

**Comment by deecodeme** [comment]
Hi @RobWin 

I hope you are doing well. The review is still pending, please check at your time convenience. Thanks  
https://github.com/resilience4j/resilience4j/pull/1991
